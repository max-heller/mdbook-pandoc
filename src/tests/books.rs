use std::path::{Path, PathBuf};

use once_cell::sync::Lazy;

use super::{Config, MDBook};

static BOOKS: Lazy<PathBuf> = Lazy::new(|| Path::new(env!("CARGO_MANIFEST_DIR")).join("books"));

#[test]
#[ignore]
fn mdbook_guide() {
    let logs = MDBook::load(BOOKS.join("mdBook/guide"))
        .config(Config {
            hosted_html: Some("https://rust-lang.github.io/mdBook/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn cargo_book() {
    let logs = MDBook::options()
        .max_log_level(tracing::Level::DEBUG)
        .load(BOOKS.join("cargo/src/doc"))
        .config(Config {
            hosted_html: Some("https://doc.rust-lang.org/cargo/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn rust_book() {
    let logs = MDBook::load(BOOKS.join("rust-book"))
        .config(Config {
            hosted_html: Some("https://doc.rust-lang.org/book/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn nomicon() {
    let logs = MDBook::load(BOOKS.join("nomicon"))
        .config(Config {
            hosted_html: Some("https://doc.rust-lang.org/nomicon/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn rust_by_example() {
    let logs = MDBook::load(BOOKS.join("rust-by-example"))
        .config(Config {
            hosted_html: Some("https://doc.rust-lang.org/rust-by-example/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn rust_edition_guide() {
    let logs = MDBook::load(BOOKS.join("rust-edition-guide"))
        .config(Config {
            hosted_html: Some("https://doc.rust-lang.org/edition-guide/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn rust_embedded() {
    let logs = MDBook::load(BOOKS.join("rust-embedded"))
        .config(Config {
            hosted_html: Some("https://docs.rust-embedded.org/book/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn rust_reference() {
    let logs = MDBook::load(BOOKS.join("rust-reference"))
        .config(Config {
            hosted_html: Some("https://doc.rust-lang.org/reference/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}

#[test]
#[ignore]
fn rustc_dev_guide() {
    let logs = MDBook::load(BOOKS.join("rustc-dev-guide"))
        .config(Config {
            hosted_html: Some("https://rustc-dev-guide.rust-lang.org/".into()),
            ..Config::pdf()
        })
        .build()
        .logs;
    insta::assert_snapshot!(logs);
}
