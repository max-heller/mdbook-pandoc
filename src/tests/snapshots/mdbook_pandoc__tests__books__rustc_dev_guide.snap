---
source: src/tests/books.rs
expression: logs
snapshot_kind: text
---
 WARN mdbook::preprocess::cmd: The command wasn't found, is the "mermaid" preprocessor installed?    
 WARN mdbook::preprocess::cmd: 	Command: mdbook-mermaid    
 INFO mdbook::book: Running the pandoc backend    
 INFO mdbook_pandoc::pandoc::renderer: Running pandoc    
[WARNING] [makePDF] LaTeX Warning: Label
  `book__pandoc__pdf__src__traits__goals-and-clauses.md__domain-goals'
  multiply defined.
[WARNING] [makePDF] LaTeX Warning: Label
  `book__pandoc__pdf__src__borrow_check__region_inference.md__region-variables'
  multiply defined.
[WARNING] [makePDF] LaTeX Warning: Hyper reference
  `book__pandoc__pdf__src__building__suggested.md__faster-builds-with---keep-stage'
  on $PAGE undefined on input $LINE.
[WARNING] [makePDF] LaTeX Warning: Hyper reference
  `book__pandoc__pdf__src__building__how-to-build-and-run.md__toolchain'
  on $PAGE undefined on input $LINE.
[WARNING] [makePDF] LaTeX Warning: Hyper reference
  `book__pandoc__pdf__src__parallel-rustc.md__Codegen' on $PAGE undefined on input $LINE.
[WARNING] [makePDF] LaTeX Warning: Hyper reference
  `book__pandoc__pdf__src__hir.md__the-hir-map' on $PAGE
  undefined on input $LINE.
[WARNING] [makePDF] LaTeX Warning: Hyper reference
  `book__pandoc__pdf__src__query.md__adding-a-new-kind-of-query'
  on $PAGE undefined on input $LINE.
[WARNING] [makePDF] LaTeX Warning: Hyper reference
  `book__pandoc__pdf__src__diagnostics.md__future-incompatible'
  on $PAGE undefined on input $LINE.
[WARNING] [makePDF] LaTeX Warning: Hyper reference
  `book__pandoc__pdf__src__ty_module__generic_arguments.md__genericargkind'
  on $PAGE undefined on input $LINE.
[WARNING] [makePDF] LaTeX Warning: Hyper reference
  `book__pandoc__pdf__src__llvm-coverage-instrumentation.md__recommended-configtoml-settings'
  on $PAGE undefined on input $LINE.
[WARNING] [makePDF] LaTeX Warning: Hyper reference
  `book__pandoc__pdf__src__appendix__glossary.md__TyCtxt' on
  $PAGE undefined on input $LINE.
[WARNING] [makePDF] LaTeX Warning: Hyper reference
  `book__pandoc__pdf__src__appendix__background.md__what-is-a-debruijn-index'
  on $PAGE undefined on input $LINE.
[WARNING] [makePDF] LaTeX Warning: There were undefined references.
[WARNING] [makePDF] LaTeX Warning: There were multiply-defined labels.
 INFO mdbook_pandoc::pandoc::renderer: Wrote output to book/pandoc/pdf/book.pdf
